src/Plugin/Condition/AdoType.php

```diff
@@ -0,0 +1,171 @@
+ *   label = @Translation("ADO Type"),
+ *   context_definitions = {
+ *     "node" = @ContextDefinition("entity:node", label = @Translation("node"))
+ *   }
+ * )
+ */
+class AdoType extends ConditionPluginBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Creates a new AdoType instance.
+   *
+   * @param array $configuration
+   *   The plugin configuration.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form['ado_types'] = [
+      '#title' => $this->pluginDefinition['label'],
+      '#description' => t(
+        'Enter one or more ADO type names, <strong>one per line</strong>. A list of ADO types can be viewed on the <a href="@link">ADO Type to View Mode Mapping Form</a>.',
+        ['@link' => Url::fromRoute('format_strawberryfield.view_mode_mapping_settings_form')->toString()]
+      ),
+      '#type' => 'textarea',
+      '#default_value' => implode(PHP_EOL, $this->configuration['ado_types']),
+      '#required' => TRUE,
+    ];
+    $form['recurse_ado_types'] = [
+      '#title' => t("Recurse metadata"),
+      '#description' => t('Do you want this condition to look for type values everywhere in the ADO metadata, instead of just at the top level? For example, entering "Image" would cause this condition to match an ADO having an attached image file if this option is checked.'),
+      '#type' => 'checkbox',
+      '#default_value' => $this->configuration['recurse_ado_types'],
+    ];
+    return parent::buildConfigurationForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    $this->configuration['ado_types'] = array_filter(array_map('trim', explode(PHP_EOL, $form_state->getValue('ado_types'))));
+    parent::submitConfigurationForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function evaluate() {
+    // Returns true if no ado types are selected and negate option is disabled.
+    if (empty($this->configuration['ado_types']) && !$this->isNegated()) {
+      return TRUE;
+    }
+    /** @var \Drupal\node\Entity\Node $node */
+    $entity = $this->getContextValue('node');
+
+    $ado_types = [];
+    if ($sbf_fields = \Drupal::service('strawberryfield.utility')->bearsStrawberryfield($entity)) {
```

Comment:
@patdunlavey can you inject this service please? Since you are already using  ContainerFactoryPluginInterface that would be great (sorry you know how my code reviews are... so sorry)

Reply:
Re. the configs, this seems to come for free with context plugins. When I do drush cex, I get a `context.context.[context_name].yml` file. Or am I misunderstanding you?

Reply:
Is it possible that the [annotation](https://github.com/esmero/format_strawberryfield/pull/222/files#diff-779e0d5842f0dd694d46db304b24da1bccdf15d86f7053cc4279da8eda1862c1R13-R23) makes the schema definition unnecessary?